core/commands/storage/upload/upload/init.go

```diff
@@ -314,10 +316,25 @@ func downloadShardFromClient(ctxParams *uh.ContextParams, guardContract *guardpb
 		scaledRetry = highRetry
 	}
 	expir := uint64(guardContract.RentEnd.Unix())
+
+	hostPid, err := peer.IDB58Decode(guardContract.HostPid)
+	if err != nil {
+		return err
+	}
 	// based on small and large file sizes
 	err = backoff.Retry(func() error {
 		ctx, cancel := context.WithTimeout(context.Background(), scaled)
 		defer cancel()
+
+		// swarm connect to renter
+		for i := 0; i < 3; i++ {
+			go func() {
+				ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+				defer cancel()
+				ctxParams.Api.Swarm().Connect(ctx, peer.AddrInfo{ID: hostPid})
```

Comment:
suggest do this without goroutine, because if swarm connect is failing, likely you won't download shard successfully (since nobody else has shard X), you can do 3 times, if not success, return err right away (no need to perform download)